db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryCache.java

```diff
@@ -270,7 +270,7 @@ public Conglomerate conglomerateCacheFind(Long conglomId) throws StandardExcepti
     }
 
     public void conglomerateCacheAdd(Long conglomId, Conglomerate conglomerate,TransactionController xactMgr) throws StandardException {
-        if (!dd.canUseCache(xactMgr))
+        if (!dd.canUseCache(xactMgr)  && conglomId>=DataDictionary.FIRST_USER_TABLE_NUMBER)
```

Comment:
We can't rely on DataDictionary.FIRST_USER_TABLE_NUMBER to separate user from system conglomerates. When upgrading we sometimes add new system tables and that would make user tables fall inside the "system tables" category. I would add a method that checks a conglomId against the coreInfo and nonCoreInfo arrays.